### PR TITLE
Add and use `get_indices_for_slot` and `get_block_hash`

### LINKED ISSUE #59: add `get_indices_for_slot` and `get_block_hash`
# Issue
The v2.1 spec has two new helpers defined (`get_indices_for_slot` and `get_block_hash`). The current code was written before these were added.

# Suggested implementation
* add and test the helpers
* update the code anywhere the spec suggests to use these helpers

### PR DESCRIPTION
**Pull request by hwwhww**
### What was wrong?
Fixes #59

### How was it fixed?
* Rename `indices_for_heights` to `indices_for_slots`
* Added `get_indices_for_slot` and `get_block_hash` helper functions.
* Modified `get_signed_parent_hashes()` and `get_attestation_indices` to use those functions.

I think if we use the old `get_signed_parent_hashes()` + add some boundary cases checks, it would be faster and efficient than the new one.

#### TODO: Add more tests and refactor.

### CODE REVIEW

File: beacon_chain/state/helpers.py [hunk 1]
```diff
@@ -131,3 +134,26 @@ def get_new_shuffling(seed,
...
+def get_block_hash(
+        active_state,
+        current_block,
+        slot,
+        config=DEFAULT_CONFIG):
+    cycle_length = config['cycle_length']
+
+    sback = current_block.slot_number - cycle_length * 2
+    assert sback <= slot < sback + cycle_length * 2
+    return active_state.recent_block_hashes[slot - sback]
```
Comment (terenc3t) [inline beacon_chain/state/helpers.py]: have you considered the case where `sback` could be negative?
Example:
`current_block.slot_number` = 10
`sback` = 10 - 64 * 2 = -118
then it'll return `active_state.recent_block_hashes[slot + 118]`

Comment (hwwhww) [reply beacon_chain/state/helpers.py]: I think it's okay since practically:
1. `active_state.recent_block_hashes` was initialized to a list of `CYCLE_LENGTH * 2` zero hashes, and then updated via `get_new_recent_block_hashes` block by block. 
2. The domain of `slot - sback` was guaranteed by the assertion `sback <= slot < sback + cycle_length * 2`, that said, `0 <= slot - sback < CYCLE_LENGTH * 2`.

I will add a test of the scenario you mentioned, thank you! 👍

File: beacon_chain/state/helpers.py [hunk 2]
```diff
@@ -131,3 +134,26 @@ def get_new_shuffling(seed,
...
+
+
+def get_indices_for_slot(
+        crystallized_state,
+        slot,
+        config=DEFAULT_CONFIG):
+    cycle_length = config['cycle_length']
+
+    ifh_start = crystallized_state.last_state_recalc - cycle_length
+    assert ifh_start <= slot < ifh_start + cycle_length * 2
```
Comment (ChihChengLiang) [inline beacon_chain/state/helpers.py]: What does `ifh` stands for?

Comment (hwwhww) [reply beacon_chain/state/helpers.py]: Oh, `indices_for_height`, but it was renamed to `indices_for_slot`. I will update it. Thanks.
